app/src/main/kotlin/at/bitfire/davdroid/servicedetection/HomeSetRefresher.kt

```diff
@@ -159,4 +161,31 @@ class HomeSetRefresher @AssistedInject constructor(
         }
     }
 
+    /**
+     * Evaluates whether a response if personal or not.
+     * It takes the [Owner] property from the response, and compares its value against [service]'s [Service.principal].
+     *
+     * If either one of those is not set (`null`), this function returns `null`.
+     * @param davResponse The response to process.
+     * @return
+     * - `true` if the owner matches a principal.
+     * - `false` if the owner doesn't match a principal.
+     * - `null` if the owner and/or principal is not set / unknown.
+     */
+    private fun isPersonal(davResponse: Response): Boolean? {
+        // Owner must be set in order to check if the home set is personal
+        val ownerHref = davResponse[Owner::class.java]?.href
+        val principal = service.principal
+
+        // If either Owner or principal is not set, return null
```

Comment:
Comments are always good to summarize or explain, but this one strikes me as obsolete, since it just states the obvious. If it said why we return null here it would be beneficial, otherwise I would simply remove it.

I would also bail out even faster and simply do:

```
val ownerHref = davResponse[Owner::class.java]?.href 
    ?: return false
val principal = service.principal
    ?: return false
```

The comment `// Owner must be set in order to check if the home set is personal` is good and explains why we return asap already. :)